src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -0,0 +1,219 @@
+        try {
+            return Json.mapper.convertValue(imp.getExt().get("bidder"), ExtImpConsumable.class);
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+    }
+
+    private static MultiMap resolveHeaders(BidRequest request) {
+        final MultiMap headers = HttpUtil.headers();
+        final Device device = request.getDevice();
+        if (device != null) {
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, "User-Agent", device.getUa());
+            final String ip = device.getIp();
+            if (StringUtils.isNotBlank(ip)) {
+                headers.add("Forwarded", "for=" + ip);
+                headers.add("X-Forwarded-For", ip);
+            }
+        }
+
+        final User user = request.getUser();
+        if (user != null && StringUtils.isNotBlank(user.getBuyeruid())) {
+            headers.add("Cookie", String.format("azk=%s", user.getBuyeruid().trim()));
+        }
+
+        final Site site = request.getSite();
+        if (site != null && StringUtils.isNotBlank(site.getPage())) {
+            headers.set("Referer", site.getPage());
+            headers.set("Origin", HttpUtil.validateUrl(site.getPage()));
+        }
+
+        return headers;
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<ConsumableBidRequest> httpCall, BidRequest bidRequest) {
+        final ConsumableBidResponse consumableResponse;
+        try {
+            consumableResponse = Json.decodeValue(httpCall.getResponse().getBody(), ConsumableBidResponse.class);
+        } catch (DecodeException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+        }
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidderBid> bidderBids = extractBids(bidRequest, consumableResponse.getDecisions(), errors);
+        return Result.of(bidderBids, errors);
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, Map<String, ConsumableDecision> impIdToDecisions,
+                                               List<BidderError> errors) {
+        final List<BidderBid> bidderBids = new ArrayList<>();
+
+        for (Map.Entry<String, ConsumableDecision> entry : impIdToDecisions.entrySet()) {
+            final ConsumableDecision decision = entry.getValue();
+
+            final ConsumablePricing pricing = decision.getPricing();
+            if (pricing != null && pricing.getClearPrice() != null) {
+                final String impId = entry.getKey();
+                final Imp imp;
+                try {
+                    imp = getImpById(impId, bidRequest);
+                } catch (PreBidException e) {
+                    errors.add(BidderError.badServerResponse(e.getMessage()));
+                    continue;
+                }
+                final Format firstFormat = imp.getBanner().getFormat().get(0);
```

Comment:
Can `formats` be null?

Reply:
since they don't have a logic for this case in Go, I'll check formats for being empty, and if they are - add an error and skip current iteration.